src/App.tsx

```diff
@@ -15,71 +16,87 @@ export const goods = [
   'Garlic',
 ];
 
-export const App: React.FC = () => (
-  <main className="section container">
-    <h1 className="title">No goods selected</h1>
+type State = {
+  selectedGood: string,
+};
 
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
+export class App extends React.Component <{}, State> {
+  state:Readonly<State> = {
+    selectedGood: 'Jam',
+  };
 
-      {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-      <button
-        data-cy="ClearButton"
-        type="button"
-        className="delete ml-3"
-      />
-    </h1>
+  handler = (good: string) => (
+    this.state.selectedGood === good
+      ? this.setState({ selectedGood: '' })
+      : this.setState({ selectedGood: good })
+  );
 
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+  render() {
+    const { selectedGood } = this.state;
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
+    return (
+      <main className="section container">
+        <div>
+          {selectedGood
+            ? (
+              <h1 className="title is-flex is-align-items-center">
+                {`${selectedGood} is selected`}
+                {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
+                <button
+                  data-cy="ClearButton"
+                  type="button"
+                  className="delete ml-3"
+                  onClick={() => this.setState({ selectedGood: '' })}
+                />
+              </h1>
+            )
+            : <h1 className="title">No goods selected</h1>}
+        </div>
+        <table className="table">
+          <tbody>
+            <div>
+              {goods.map(good => {
+                return (
+                  <tr
+                    data-cy="Good"
+                    className={classNames(
+                      {
+                        'has-background-success-light': selectedGood === good,
+                      },
+                    )}
+                  >
+                    <td>
+                      <button
+                        id={good}
+                        data-cy="AddButton"
```

Comment:
```suggestion
                        data-cy={
                          selectedGood === good
                            ? 'RemoveButton'
                            : 'AddButton'
                        }
```

Тесты не проходят именно из-за этого)